resolver.go

```diff
@@ -0,0 +1,45 @@
+package doh
+
+import (
+	"context"
+	"net"
+	"strings"
+
+	madns "github.com/multiformats/go-multiaddr-dns"
+)
+
+type Resolver struct {
+	url string
+}
+
+func NewResolver(url string) *Resolver {
+	if !strings.HasPrefix(url, "https://") {
+		url = "https://" + url
+	}
+	if !strings.HasSuffix(url, "/dns-query") {
+		url = url + "/dns-query"
+	}
+
+	return &Resolver{url: url}
+}
+
+var _ madns.BasicResolver = (*Resolver)(nil)
+
+func (r *Resolver) LookupIPAddr(ctx context.Context, domain string) ([]net.IPAddr, error) {
+	ip4, err := doRequestA(ctx, r.url, domain)
+	if err != nil {
+		return nil, err
+	}
+
+	ip6, err := doRequestAAAA(ctx, r.url, domain)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Can we do those in parallel? 

On my fiber each curl request is between 150-250ms, and we could save some time here.

Reply:
Sure, we can spawn a couple of goroutines.

Reply:
done.